### PR TITLE
Delete ConfigManager class and replace its valid usages

### LINKED ISSUE #177: Refactor - Excise ConfigManager from MetamaskController
generic reactive store (observable)
break off actions
means of merging stores

### PR DESCRIPTION
**Pull request by whymarrh**
Closes #177

This PR delete the `ConfigManager` and replaces its valid usages with `PreferencesController`—only two of the config properties are actually used (`seedWords` and `forgottenPassword`). I've left `lostAccounts` hard-coded as an empty array since we aren't tracking them anymore and since the migration that caused lost accounts<sup>[\[1\]](https://medium.com/metamask/metamask-3-migration-guide-914b79533cdd)</sup> was a good few versions ago.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -314,18 +308,15 @@ module.exports = class MetamaskController extends EventEmitter {
...
   getState () {
-    const wallet = this.configManager.getWallet()
     const vault = this.keyringController.store.getState().vault
-    const isInitialized = (!!wallet || !!vault)
+    const isInitialized = !!vault
 
     return {
       ...{ isInitialized },
       ...this.memStore.getFlatState(),
-      ...this.configManager.getConfig(),
```
Comment (whymarrh) [inline app/scripts/metamask-controller.js]: I found this line to be particularly egregious, because it always returned an empty object 😔

Comment (danfinlay) [reply app/scripts/metamask-controller.js]: 🤦‍♂️

**Review by danfinlay — APPROVED** [review]
Surprisingly simple change to resolve a very long-running technical debt.  Thank you, Whymarr!

**Comment by kumavis** [comment]
So much red delicious! Great work @whymarrh

**Pull request merged**